src/main/java/com/diffblue/deeptestutils/Reflector.java

```diff
@@ -324,16 +324,28 @@ private static void makePublic(final CtClass c) {
   }
 
   /**
-   * This forces the creation of an instance for a given class. If the
-   * class provides a public default constructor, it is called. If the class has
-   * a private default constructor, it is made accessible and is then called.
+   * This forces the creation of an instance for a given class. If the class
+   * provides a public default constructor, it is called. If the class has a
+   * private default constructor, it is made accessible and is then called. Else
+   * we use Objenesis to force creation of an instance.
+   *
+   * There are several exceptions that `Constructor.netInstance` can throw. We
+   * catch `IllegalAccessException` and `InstantiationException` and continue
+   * with Objenesis in those cases. `IllegalArgumentException` cannot be thrown
+   * as we only try to use the default constructor which has no parameters. An
+   * `InvocationTargetException` signals an exception in the constructor and is
+   * passed to the calling method, just as `ExceptionInInitializerError`.
    *
    * @param <T> type parameter of the class
    * @param cl a <code>Class</code> the class to instantiate
    * @return an <code>Object</code> which is an instance of the specified class
+   *
+   * @throws InvocationTargetException if the default constructor throws an
+   * exception
    */
   @SuppressWarnings("unchecked")
-  public static <T> T getInstance(final Class<T> cl) {
+  public static <T> T getInstance(final Class<T> cl)
```

Comment:
Why do the two `getInstance` methods have completely separate code??

Reply:
one takes a `String` argument, tries to get a corresponding `Class` object or construct it if necessary and then calls the `getInstance` with `Class` argument.

Reply:
the `Class` variant tries to somehow create an Object of that type